cmd/vendor/golang.org/x/arch: add disasm support for Zicbom and Zicboz extensions

ctk-1998 opened the pull request:
<!-- 
+ The PR title is formatted as follows: `net/http: frob the quux before blarfing`
  + The package name goes before the colon
  + The part after the colon uses the verb tense + phrase that completes the blank in,
    "This change modifies Go to ___________"
  + Lowercase verb after the colon
  + No trailing period
  + Keep the title as short as possible. ideally under 76 characters or shorter 
-->

## Related Issue(s) & Descriptions
issue: #53 

<!-- List related issues and pull requests. If this PR fixes an issue, please add it using Fixes #????  -->

## Checklist

-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

newborn22 committed:
CI: support basic github CI (#4)

* basic ci

* arm runner

* fix arm64 github runnner

* delete all.bash test in rv64

* add test codegen

* update jobs name

* modify pr template

* fix app test and go vet

* fix go vet and app test

lxq015 committed:
cmd/internal/obj/riscv: add assembly support of Zicond extension

* cmd/internal/obj/riscv: add assembly support of Zicond extension

This patch adds the assembly support of Zicond extension, which is
defined in chapter 12 of RISC-V ISA manual.

Zicond adds two instructions to do integer conditional operations
and is mandatory in RVA23 profile.

newborn22 committed:
Contribution guidelines for Go RISC-V Repository (PR #6)

* contribution guidlines

* contribution guidlines CN

* update some content

* fix content

* fix content

* fix content

newborn22 committed:
CI: go vet workflow: ignore errors of unchanged files in go vet (#12)

* go vet CI: ignore errors in unchanged files

* remove unimportant output

* remove useless else

lxq015 committed:
cmd/compile/internal/ssa: add codegen for Zicond extension (#8)

* cmd/compile/internal/ssa: add codegen for Zicond extension

This patch adds code generation support for the Zicond extension.

We convert branches into CondSelect operations in branchelim and rewrite them into Zicond instructions.

Additionally, optimization rules have been added based on the riscv-isa-manual.

lxq015 committed:
cmd/asm, cmd/internal/obj: add crypto extension for riscv64 (#14)

This CL adds assembly support for the RISC‑V Vector Cryptography extensions
Zvbb, Zvbc, Zvkb, Zvkg, Zvkned, Zvknh[ab], Zvksed, Zvksh
#18

newborn22 committed:
cmd/asm, cmd/internal/obj: add riscv64 generic CSR ops (#25)

Support is added for the generic RISC-V CSR operations; CSRRC, CSRRCI,
CSRRS, CSRRSI, CSRRW, CSRRWI.  These instructions require special
handling as their second operand is a symbolic CSR register name and
not an immediate value or a register.  CSR names are implemented as
special operands.

RISC-V CSRs are not currently saved and restored when a go routine is
asynchronously pre-empted so it is only safe to use these instructions
in hand written assembler.  Note that CSRRS was already partially
supported by the assembler so this restriction predates this commit.
We mention it here as this commit makes CSRRS much easier to use.

Change-Id: I9ff8d804328b418a879d463e7d9cc31f489c7a00

Co-authored-by: Mark Ryan <markdryan@rivosinc.com>

newborn22 committed:
cmd/internal/obj/riscv: implement unit-stride vector segment load/store instructions (#13)

* add assembly support for unit-stride vector segment load/store

newborn22 committed:
support stride vector segment load/store in assembly (#20)

newborn22 committed:
support index vector segment load/store in assembly (#22)

newborn22 committed:
update readme to go1.25.1

newborn22 committed:
support vector unit-stride fault-only-first load instructions in assembly level (#24)

newborn22 committed:
cmd/asm, cmd/internal/obj: add riscv64 pseudo CSR ops (#28)

Add support for the riscv64 pseudo CSR ops; CSRR, CSRW, CSRS, CSRC,
CSRWI, CSRSI, CSRCI.

We also add tests to check that we can correctly encode CSRR
instructions that read all the RVV CSR registers.

Change-Id: I50b2d6da4f5847e65ab8017cd36f068b2ce525b3

Co-authored-by: Mark Ryan <markdryan@rivosinc.com>

wenchangping committed:
cmd/asm: add zimop extension for riscv64(#16) (#21)

* cmd/asm: add zimop extension for riscv64(#16)

* cmd/asm: add zimop extension for riscv64(#16)

* cmd/asm: add zimop extension for riscv64(#16)

---------

Co-authored-by: Qiuweihong <953950914@qq.com>
Co-authored-by: 文昌平10295505 <wen.changping@zte.com.cn>

ctk-1998 committed:
cmd/asm: add Zicfiss extension for riscv64 (#31)

* Fix Zicfiss asm based on Zimop

* fix bug

* move Zicfiss pseudo-instructions rewriting

* remove disasm

---------

Co-authored-by: cuitiankai10342768 <cui.tiankai@zte.com.cn>

wenchangping committed:
add support for Zihintpause extension on riscv64 #36 (#37)

Co-authored-by: 文昌平10295505 <wen.changping@zte.com.cn>

ctk-1998 committed:
add asm support for Zabha (#39)

Co-authored-by: cuitiankai10342768 <cui.tiankai@zte.com.cn>

newborn22 committed:
cmd/internal/obj/riscv: add encoding for compressed riscv64 instructions (#32)

Regenerate instruction encodings for riscv64, including rv64_c, rv_c and rv_c_d.

Updates #71105

Change-Id: I004cc00290bf018ebb848592d84d924481f03851
Reviewed-on: https://go-review.googlesource.com/c/go/+/523476
Reviewed-by: Meng Zhuo <mengzhuo1203@gmail.com>
Reviewed-by: Cherry Mui <cherryyz@google.com>
LUCI-TryBot-Result: Go LUCI <golang-scoped@luci-project-accounts.iam.gserviceaccount.com>
Reviewed-by: Dmitri Shuralyov <dmitshur@google.com>

Co-authored-by: Joel Sing <joel@sing.id.au>

ctk-1998 committed:
add disasm support for Zabha (#43)

Co-authored-by: cuitiankai10342768 <cui.tiankai@zte.com.cn>

newborn22 committed:
CI: auto delete branch on merge (#41)

newborn22 committed:
support zawrs assembly (#47)

Commit:
cmd/vendor/golang.org/x/arch: add disasm support for Zicbom and Zicboz extensions

ctk-1998 committed:
Merge branch 'go1.25.3-zte-dev' into Zicbom_Zicboz_disasm

wangpc-pp commented on src/cmd/asm/internal/asm/testdata/riscv64.s:
This is Zihintntl not CMO extensions.

ctk-1998 replied:
<img width="1124" height="959" alt="image" src="https://github.com/user-attachments/assets/9210d83e-7020-4d7a-8280-632deffd9f19" />

We just modified the base from go1.25.1-zte-dev to go1.25.3-zte-dev. I have rebased it and it should be correct now.

ctk-1998 commented:
> This is Zihintntl not CMO extensions.

@wangpc-pp 
We just modified the base from go1.25.1-zte-dev to go1.25.3-zte-dev. I have rebased two prs and they should be correct now.

wenchangping reviewed (approved):
+1

wangpc-pp reviewed (approved):
+2

lxq015 reviewed (approved):
+1

newborn22 reviewed (approved):
+2
